File: SearchNShop_and/app/src/main/java/co/esclub/searchnshop/model/ShopItem.java
package co.esclub.searchnshop.model;

import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.NotNull;

import io.realm.RealmObject;

/**
 * Created by tae.kim on 06/06/2017.
 */

public class ShopItem extends RealmObject {
    public ShopItem() {

    }

    public ShopItem(@NotNull String title,
                    @NotNull String link,
                    @NotNull String image,
                    @NotNull String mallName, int position) {
        this.title = title;
        this.link = link;
        this.image = image;
        this.mallName = mallName;
        this.position = position;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getMallName() {
        return mallName;
    }

    public void setMallName(String mallName) {
        this.mallName = mallName;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    @SerializedName("title")
    private String title;
    @SerializedName("link")
    private String link;
    @SerializedName("image")
    private String image;
    @SerializedName("mallName")
    private String mallName;
    private int position;
}

File: SearchNShop_and/app/src/main/java/co/esclub/searchnshop/model/item/ShopItem.kt
package co.esclub.searchnshop.model.item

import co.esclub.searchnshop.model.firebase.ShopItem
import com.google.gson.annotations.SerializedName
import io.realm.DynamicRealmObject
import io.realm.RealmObject

/**
 * Created by tae.kim on 06/06/2017.
 */


open class ShopItem : RealmObject, Item {
    constructor() {

    }

    constructor(title: String,
                link: String,
                image: String,
                mallName: String, position: Int) {
        this.title = title
        this.link = link
        this.image = image
        this.mallName = mallName
        this.position = position
    }

    constructor(item: ShopItem) {
        this.title = item.title
        this.link = item.link
        this.image = item.image
        this.mallName = item.mallName
        this.position = item.position
    }

    @SerializedName("title")
    open var title: String? = null
    @SerializedName("link")
    open var link: String? = null
    @com.google.gson.annotations.SerializedName("image")
    open var image: String? = null
    @com.google.gson.annotations.SerializedName("mallName")
    open var mallName: String? = null
    open var position: Int = 0

    constructor(item: DynamicRealmObject) {
        this.title = item.getString("title")
        this.link = item.getString("link")
        this.image = item.getString("image")
        this.mallName = item.getString("mallName")
        this.position = item.getInt("position")
    }
}
